pdr_backend/contract/base_contract.py

```diff
@@ -14,14 +14,17 @@ def __init__(self, web3_pp, address: str, contract_name: str):
         if not isinstance(web3_pp, Web3PP):
             raise ValueError(f"web3_pp is {web3_pp.__class__}, not Web3PP")
         self.web3_pp = web3_pp
-        self.config = web3_pp.web3_config  # for convenience
         self.contract_address = self.config.w3.to_checksum_address(address)
         self.contract_instance = self.config.w3.eth.contract(
             address=self.config.w3.to_checksum_address(address),
             abi=web3_pp.get_contract_abi(contract_name),
         )
         self.contract_name = contract_name
 
+    @property
```

Comment:
@trizin we need to do this as a property, otherwise it's dangerous because we have two separate references to web3_config (one via web_3pp, one via config)